storey/aggregations.py

```diff
@@ -126,13 +127,40 @@ async def _do(self, event):
             if isinstance(self._emit_policy, EmitEveryEvent):
                 await self._emit_event(key, event)
             elif isinstance(self._emit_policy, EmitAfterMaxEvent):
-                self._events_in_batch[key] = self._events_in_batch.get(key, 0) + 1
-                if self._events_in_batch[key] == self._emit_policy.max_events:
+                if key in self._events_in_batch:
+                    self._events_in_batch[key]['counter'] = self._events_in_batch[key].get('counter') + 1
+                else:
+                    self._events_in_batch[key] = {}
+                    self._events_in_batch[key]['counter'] = 1
+                if self._events_in_batch[key]['counter'] == self._emit_policy.max_events:
                     await self._emit_event(key, event)
-                    self._events_in_batch[key] = 0
+                    self._events_in_batch[key]['counter'] = 0
+            elif isinstance(self._emit_policy, EmitAfterDelay):
+                if key not in self._events_in_batch:
+                    self._events_in_batch[key] = {}
+                self._events_in_batch[key]['event'] = event
+                self._events_in_batch[key]['time'] = event.time
+                if self._timeout_task is None:
+                    self._timeout_task = asyncio.get_running_loop().create_task(self._sleep_and_emit())
+
         except Exception as ex:
             raise ex
 
+    async def _sleep_and_emit(self):
+        while self._events_in_batch:
+            key = next(iter(self._events_in_batch.keys()))
+            if self._events_in_batch[key]['event'] is None:
+                continue
+            time_delta = datetime.now() - self._events_in_batch[key]['time']
```

Comment:
You are waiting based on the time of the first event but on line 141-2 you keep overriding the event. does that makes sense?

Reply:
event time != system time
The two should not be mixed.

Reply:
Fixed Batching in #134